### ISSUE TITLE
Better error message on import when missing common dependencies

### ISSUE THREAD

**Issue by horta** [dask/dask#3740]
Steps to produce the error:

- docker run -it ubuntu:latest /bin/bash
- apt-get update --yes && apt-get upgrade --yes && apt-get install git build-essential curl wget --yes
- wget https://repo.continuum.io/miniconda/Miniconda3-latest-Linux-x86_64.sh
- bash Miniconda3-latest-Linux-x86_64.sh -b
- export PATH=/root/miniconda3/bin:$PATH
- hash -r
- pip install dask
- python -c "import dask.array"

Output:

```
Traceback (most recent call last):
  File "/root/miniconda3/lib/python3.6/site-packages/dask/array/core.py", line 21, in <module>
    from cytoolz import (partition, concat, join, first,
ModuleNotFoundError: No module named 'cytoolz'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "<string>", line 1, in <module>
  File "/root/miniconda3/lib/python3.6/site-packages/dask/array/__init__.py", line 4, in <module>
    from .core import (Array, block, concatenate, stack, from_array, store,
  File "/root/miniconda3/lib/python3.6/site-packages/dask/array/core.py", line 26, in <module>
    from toolz import (partition, concat, join, first,
ModuleNotFoundError: No module named 'toolz'
```

**Comment by jcrist** [comment]
You probably want `pip install dask[array]` or `pip install dask[complete]`. See http://dask.pydata.org/en/latest/install.html#pip.

**Comment by horta** [comment]
I see. But should it not give a more meaninful error?

**Comment by jcrist** [comment]
Sure. That'd be a good issue for the scipy sprint this weekend (or any new contributor :) ).

The code should look similar to that for `dask.distributed` (https://github.com/dask/dask/blob/master/dask/distributed.py#L6-L11).

Should look something like a try-block around all statements in `__init__`, and on `ImportError` raise a new `ImportError` explaining the common install issue here (users installing `pip install dask` instead of `pip install[something]`), with the original error message concatenated at the end.

We'd want to do this for `dask.array`, `dask.dataframe`, `dask.bag`, and `dask.delayed` (around the import for toolz) - this mirrors the structure in `extras_require`: https://github.com/dask/dask/blob/master/setup.py#L10-L17.

**Comment by jcrist** [comment]
Fixed by #3771. Closing.
